examples/transient/transient_ex3/advection_system.C

```diff
@@ -0,0 +1,137 @@
+// local includes
+#include "advection_system.h"
+
+// LibMesh includes
+#include "libmesh/equation_systems.h" // EquationSystems::comm()
+#include "libmesh/getpot.h" // GetPot input file parsing
+#include "libmesh/libmesh_logging.h" // LOG_SCOPE
+#include "libmesh/numeric_vector.h"
+#include "libmesh/sparse_matrix.h"
+#include "libmesh/string_to_enum.h"
+
+namespace libMesh
+{
+
+AdvectionSystem::AdvectionSystem (EquationSystems& es,
+		                  const std::string& name,
+		                  const unsigned int number)
+  : Parent(es, name, number),
+    _q1_var(0),
+    _fe_order(CONSTANT),
+    _fe_family(MONOMIAL)
+{
+  // Allocate NumericVectors in _Fh. I could not figure out
+  // how to do this in the initialization list, I don't think it's
+  // possible.
+  // TODO: the number of problem dimensions is hard-coded here, we should
+  // make this depend on the input file parameters instead.
+  for (unsigned int i=0; i<2; ++i)
+    _Fh.push_back(NumericVector<Number>::build(es.comm()));
```

Comment:
Would it make sense to postpone this until init_data()?

Reply:
Yeah, I think that makes sense. I did the update in [05ef170](https://github.com/libMesh/libmesh/pull/4073/commits/05ef17074d7135c2b43ab1e687b2973192456328) and likewise for the extra `SparseMatrices` in 
24a5ac0.